### PR TITLE
Add team icon

### PR DESCRIPTION
**Pull request by xsyann**
Add a team icon next to the username and in the user list:

Screenshot 2022-09-09 at 11 44 41

Screenshot 2022-09-09 at 11 44 53

### CODE REVIEW

File: core/client/ui/user-list.hbs.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
       <span class="js-profile">{{user.profile.name}}</span>
     </div>
     <div class="guild">
+      <img class="icon" width="{{guildIcon.width}}" height="{{guildIcon.height}}" src="{{concat '/api/files/' guildIcon.fileId}}" alt="guild-icon" />
```
Comment (ramnes) [inline core/client/ui/user-list.hbs.html]: What if `guildIcon === undefined`?

File: core/client/scenes/scene-ui.js [hunk 2]
```diff
@@ -106,6 +106,13 @@ UIScene = new Phaser.Class({
     });
   },
 
+  updateUserIcon(userId, icon) {
+    const textInstance = this.characterNamesObjects[userId];
+    if (textInstance) {
```
Comment (ramnes) [inline core/client/scenes/scene-ui.js]: Why is that check necessary?

File: core/client/user-manager.js [hunk 3]
```diff
@@ -68,7 +68,10 @@ userManager = {
     this.characters[user._id] = character;
 
     if (guest) character.updateSkin(guestSkin()); // init with custom skin
-    else character.setName(name, baseline, nameColor);
+    else {
+      character.setName(name, baseline, nameColor);
+      if (user.guildId) character.setIcon(Guilds.findOne({ _id: user.guildId })?.icon);
```
Comment (ramnes) [inline core/client/user-manager.js]: Same here, what if the guild doesn't have an icon?

File: core/server/users.js [hunk 4]
```diff
@@ -18,7 +18,11 @@ Meteor.publish('users', function (levelId) {
     };
   }
 
-  return Meteor.users.find(filters, { fields: mainFields });
+  const users = Meteor.users.find(filters, { fields: mainFields });
+  const guildIds = users.map(u => u.guildId).filter(Boolean);
+  const guilds = Guilds.find({ _id: { $in: [...new Set(guildIds)] } });
+
+  return [users, guilds];
```
Comment (ramnes) [inline core/server/users.js:25]: Why do we need this?

File: core/client/scenes/scene-ui.js (hunk 2 above)
Comment (xsyann) [reply core/client/scenes/scene-ui.js]: Theoretically it shouldn't be false, I've let it as a safeguard, if for some reasons updateUserIcon is called before a name was set.

File: core/client/user-manager.js (hunk 3 above)
Comment (xsyann) [reply core/client/user-manager.js]: It works but I agree that it makes a useless call to `setIcon` with an `undefined` param, whereas there is no need to destroy an existing icon as this method is called for new users.

File: core/server/users.js (hunk 4 above)
Comment (xsyann) [reply core/server/users.js:25]: Without publishing the guilds, it wouldn't be possible to request the Guilds collection on the client side (`Guilds.findOne({ _id: user.guildId })?.icon` in `user-manager.js` l.74)

**Pull request merged**